platforms/core-runtime/process-services/src/main/java/org/gradle/process/internal/ClientExecHandleBuilder.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.process.internal;
+
+import org.gradle.api.NonNullApi;
+import org.gradle.process.CommandLineArgumentProvider;
+import org.gradle.process.ProcessForkOptions;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.Map;
+
+@NonNullApi
+public interface ClientExecHandleBuilder extends BaseExecHandleBuilder {
```

Comment:
💭 This is a new replacement for `ExecHandleBuilder`. We can rename it to `ExecHandleBuilder` in Gradle 9 and remove current `ExecHandleBuilder`

Reply:
❗ Let's add this as a `// TODO` comment in the code.